src/NHibernate/Hql/Ast/ANTLR/Tree/BinaryLogicOperatorNode.cs

```diff
@@ -55,29 +55,78 @@ public virtual void Initialize()
 			IType lhsType = ExtractDataType( lhs );
 			IType rhsType = ExtractDataType( rhs );
 
-			if ( lhsType == null ) 
+			if (!AdjustParameterTypesForSpecificOperators(lhs,rhs,lhsType, rhsType))
 			{
-				lhsType = rhsType;
-			}
-			if ( rhsType == null ) 
-			{
-				rhsType = lhsType;
-			}
 
-			var lshExpectedTypeAwareNode = lhs as IExpectedTypeAwareNode;
-			if (lshExpectedTypeAwareNode != null)
-			{
-				lshExpectedTypeAwareNode.ExpectedType = rhsType;
-			}
-			var rshExpectedTypeAwareNode = rhs as IExpectedTypeAwareNode;
-			if (rshExpectedTypeAwareNode != null)
-			{
-				rshExpectedTypeAwareNode.ExpectedType = lhsType;
+				if (lhsType == null)
+				{
+					lhsType = rhsType;
+				}
+				if (rhsType == null)
+				{
+					rhsType = lhsType;
+				}
+
+
+				var lshExpectedTypeAwareNode = lhs as IExpectedTypeAwareNode;
+				if (lshExpectedTypeAwareNode != null)
+				{
+					lshExpectedTypeAwareNode.ExpectedType = rhsType;
+				}
+				var rshExpectedTypeAwareNode = rhs as IExpectedTypeAwareNode;
+				if (rshExpectedTypeAwareNode != null)
+				{
+					rshExpectedTypeAwareNode.ExpectedType = lhsType;
+				}
 			}
 
 			MutateRowValueConstructorSyntaxesIfNecessary( lhsType, rhsType );
 		}
 
+		private bool AdjustParameterTypesForSpecificOperators(IASTNode lhs, IASTNode rhs, IType lhsType, IType rhsType)
+		{
+			var lhsParameter = lhs as ParameterNode;
+			var rhsParameter = rhs as ParameterNode;
+			//For LIKE expressions the SqlType of the parameter should not not be controlled by the other operand.
```

Comment:
I don't understand what you mean with modifying the interface - the SetExpectedType() I proposed was intended as a private method only called from Initialize().  Though it could of course be protected and overriden if we want to do a LikeOperatorNode as a subclass, but I don't think we need a subclass just for this.